### ISSUE TITLE
[V3 Core] Set nickname not changing

### ISSUE THREAD

**Issue by TheWyn** [Cog-Creators/Red-DiscordBot#1493]
Please be sure to read through other issues as well to make sure what you are suggesting/reporting has not already
been suggested/reported

### Type:

- [ ] Suggestion
- [X] Bug

### Brief description of the problem

On 3.0.0b11, when using `[p]set nickname Test`, the bot's nickname doesn't change to Test.  Running Red in debug mode doesn't show any error's.

When using `[p]set` by itself, it outputs the following error:
![](https://cdn.discordapp.com/attachments/169868846759542785/430555335506395146/Capture.PNG)

**Comment by palmtree5** [comment]
ya, both issues confirmed. Will fix for next release

**Comment by TheWyn** [comment]
Adding onto this bug if a `[p]set` sub-command is entered incorrectly a trace is given instead of the content of `[p]help set`.

Trigger this by doing `[p]set nick test` (nickname is correct command) and receive following as trace:

```
Exception in command 'set'
Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/discord/ext/commands/core.py", line 62, in wrapped
    ret = yield from coro(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/redbot/core/core_commands.py", line 544, in _set
    admin_role.name, mod_role.name, locale
AttributeError: 'NoneType' object has no attribute 'name'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.6/site-packages/discord/ext/commands/bot.py", line 886, in invoke
    yield from ctx.command.invoke(ctx)
  File "/usr/lib/python3.6/site-packages/discord/ext/commands/core.py", line 923, in invoke
    yield from injected(*ctx.args, **ctx.kwargs)
  File "/usr/lib/python3.6/site-packages/discord/ext/commands/core.py", line 71, in wrapped
    raise CommandInvokeError(e) from e
discord.ext.commands.errors.CommandInvokeError: Command raised an exception: AttributeError: 'NoneType' object has no attribute 'name'
```

**Comment by Kowlin** [comment]
Fixed in #1494
